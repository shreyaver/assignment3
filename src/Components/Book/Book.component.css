.book {
    width: 30%;
    margin-top: 3%; 
    margin-bottom: 1.5%;
    min-width: 300px;
    max-width: 400px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}
.imagePanel {
    width:100%;
    flex-shrink: 0;
    flex-grow: 0;
    height: 300px;
    background-repeat: no-repeat;
    background-size: cover;
}
.bookData {
    background-color: white;
    font-size: 1.5rem;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 10px;
}
button {
    height: 25px;
    width:25px;
    border: none;
}
button:focus {
    outline: none;
}
.bookName {
    text-align: center;
    text-transform: uppercase;
}
.heartIcon {
    float:right;
    background-image: url("../../Icons/heart-black.svg");
}
.likedHeartIcon {
    float:right;
    background-image: url("../../Icons/heart-red.svg");
}
