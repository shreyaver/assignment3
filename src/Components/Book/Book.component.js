import React from 'react';
import './Book.component.css';
import PropTypes from 'prop-types';

const Book = (props) => {
  const imageData = require('../../Images/cover_image.jpeg');
  const { book } = props;
  let likeButton;
  if (book.rating >= 4.5) {
    likeButton = <button type="button" className="likedHeartIcon" />;
  } else {
    likeButton = <button type="button" className="heartIcon" />;
  }
  return (
    <div className="book">
      <div className="imagePanel" style={{ backgroundImage: `url(${imageData})` }} />
      <div className="bookData">
        <p className="bookName">{book.Name}</p>
        <span>
          {book.rating}
          {likeButton}
        </span>
      </div>
    </div>
  );
};
Book.propTypes = {
  book: PropTypes.shape({
    Name: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    Author: PropTypes.string,
    id: PropTypes.number,
  }).isRequired,
};
export default Book;
