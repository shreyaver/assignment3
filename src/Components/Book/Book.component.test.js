import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Book from './Book.component';
import mockData from '../../mockData';

describe('Book', () => {
  it('renders without crashing', () => {
    const book = renderer.create(<Book book={mockData[Object.keys(mockData)[0]][0]} />).toJSON();
    expect(book).toMatchSnapshot();
  });
  it('sets correct class when book is liked', () => {
    const wrapper = shallow(<Book book={Object.assign({}, mockData[Object.keys(mockData)[0]][0], { rating: 4.6 })} />);
    expect(wrapper.exists('.likedHeartIcon')).toEqual(true);
  });
  it('sets correct class when book is not liked', () => {
    const wrapper = shallow(<Book book={Object.assign({}, mockData[Object.keys(mockData)[0]][0], { rating: 4.4 })} />);
    expect(wrapper.exists('.heartIcon')).toEqual(true);
  });
});
