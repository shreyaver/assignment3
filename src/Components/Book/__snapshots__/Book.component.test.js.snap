// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book renders without crashing 1`] = `
<div
  className="book"
>
  <div
    className="imagePanel"
    style={
      Object {
        "backgroundImage": "url(cover_image.jpeg)",
      }
    }
  />
  <div
    className="bookData"
  >
    <p
      className="bookName"
    >
      Harry Potter and the Deathly Hallows (Harry Potter, #7)
    </p>
    <span>
      4.62
      <button
        className="likedHeartIcon"
        type="button"
      />
    </span>
  </div>
</div>
`;
